internal/expr/expr_test.go

```diff
@@ -506,6 +554,38 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT * AS &.* FROM t",
 		prepareArgs: []any{struct{ f int }{f: 1}},
 		err:         `cannot prepare statement: cannot use anonymous struct`,
+	}, {
+		query:       "INSERT INTO person (*, postalcode) VALUES ($Person.name, $Address.id)",
+		prepareArgs: []any{Address{}, Person{}},
+		err:         "cannot prepare statement: input expression: invalid asterisk in columns: (*, postalcode) VALUES ($Person.name, $Address.id)",
```

Comment:
This is a good point, this error could be moved there. The columns of an input or output expression can be either a list of types or an asterisk.

Now that I think about it it would be very nice to have some type level representation of the difference between the two rather than just counting the number of stars.

Reply:
Upon further consideration this would require changes outside the scope of the PR. We can look into it.